src/backend/api/meals.js

```diff
@@ -2,13 +2,81 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+router.get("/", async (req, res) => {
+  const { maxPrice, availableReservations, title, dateAfter, sortKey, sortDir, dateBefore, limit } =
+    req.query;
+  const response = {
+    data: [],
+    status: 200,
+    message: "ok",
+  };
+
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    if (maxPrice && !isNaN(maxPrice)) {
+      const result = await knex("meals").select("*").where("price", "<", +maxPrice);
+      response.data = result;
+    }
```

Comment:
I see that you have nicely implemented each of the filters. However, one thing you might have missed is that the assignment requested that **these parameters can be combined**.  For example, the user should be able to query for all meals that cost less than 50, sorted by price ascending, and which will be in the future. (date after today)

This might seem challenging at first, but actually `knex` allows to do this in quite a clean manner. 🧠 

You can just start off your method with the basic query that you WOULD execute, which is:
```javascript
let resultQuery = knex("meals");
```

NOTICE how we do **not** actually call `select()` on it, or `await` it. We are just building the query for now.

In subsequent `if` statements, if a given query parameter is supplied, for instance `maxPrice`, you can add the relevant query to the current query:

```javascript
if (maxPrice && !isNaN(maxPrice)) {
    resultQuery = resultQuery.where("price", "<", +maxPrice);
 }
```

At the end of the function, once all the possible `.where()` and `orderBy()` calls were added to the request, you can just get the result by calling:

```javascript
response.data = await resultQuery.select("meals.*");
```

Notice also how with this implementation, if no query parameters are supplied, the result will just be a simple `knex("meals").select("*")`. That is exactly what we want.

In the case of the `availableReservations` parameter, that can be more tricky. One solution would be to first find the number of reservations for each meal_id in a separate query of the `reservations` table, then use that result to filter the `meals` query.